### DISCUSSION TITLE
Override service: foobar with variable?

### DISCUSSION THREAD

**Discussion by kaihendry** [serverless/serverless#10356] [Q&A]
I raised this issue here: https://forum.serverless.com/t/serverless-yml-service-override/16102

I'm working on a serverless application called `service: foobar` and instead of deploying to this stack in my dev account, I want to create my own personal stack for testing. 

I.e. I want `service: foobar-$(whoami)`. But the problem with that approach is that I have to constantly edit the serverless and remember to make the changes when I develop, as opposed to commit code. Is there some trick I am missing to override service with say an env variable?

**Comment by mnapoli** [comment]
Hi, I would encourage you to use stages instead. If you want to deploy your personal instance, you could do `serverless deploy --stage=$(whoami)`.

**Reply by kaihendry** [reply]
I'm not confident it will create a new stack name, and how do I solve all these  `A valid file to satisfy the declaration 'file(../config.yml):STAGES.hendry-pc.SECURITY.IT.SUBNET_IDS' could not be found.` type errors? Then I'd need to edit `config.yml`, no?

**Reply by mnapoli** [reply]
> I'm not confident it will create a new stack name

It will create one stack per stage.

As for the other problem I have no idea with so little information.

**Reply by pgrzesik** [reply]
Hello @kaihendry - could you share a little bit more about the problem you're facing with the mentioned error message? If you could describe the specific use case that you're addressing that would be great too, maybe we can come up with a reasonable pattern to address it.

**Reply by kaihendry** [reply]
I already use stage. I think I'll just use the service override, since with stage I'd need redefine environment variables IIUC.
